### PR TITLE
[WIP] Add `rails start` command

### PR DESCRIPTION
**Pull request by zzak**
Having built a number of NextJS apps recently, I find myself wanting to type `start` in development.

Even though `rails s` is preferred, I actually forgot what "s" stood for and was surprised it meant "server", even after all these years! (Is a Server like a Computer now? :ipad:)

This is just a first take at adding the command which just wraps the `server` command, it can even stay hidden for now. I just think that "If I want to start my rails app, I should type `rails start`" makes a lot of sense to me. 🤔 

While we're on the subject, what about `rails dev` which is essentially `RAILS_ENV=development rails start`? And then maybe `rails prod`? These 3 are very common in my workflow outside of Rails.

### CODE REVIEW

File: railties/test/commands/start_test.rb [hunk 1]
```diff
@@ -0,0 +1,310 @@
+# frozen_string_literal: true
+
+require "isolation/abstract_unit"
+require "env_helpers"
+require "rails/command"
+require "rails/commands/start/start_command"
+
+class Rails::Command::ServerCommandTest < ActiveSupport::TestCase
```
Comment (ghiculescu) [inline railties/test/commands/start_test.rb:8]:
```suggestion
class Rails::Command::StartCommandTest < ActiveSupport::TestCase
```

**Comment by p8** [comment]
Been thinking about this as well recently...
For more complex apps I use Foreman: https://ddollar.github.io/foreman/
But I also use spring, so that is running as well.
So Foreman starts Puma with Spring enabled... that seems a bit much.

I'd expect a `rails start` to work similar to Foreman and besides starting `puma`, also starting `webpack-dev-server`, `sidekiq` and maybe `elastic-search`?
And would this also start spring for testing and maybe the console?

There is a server Railtie block to start extra processes: https://github.com/rails/rails/pull/39953

**Comment by rafaelfranca** [comment]
I'm not too keen in the start command. I really don't know what it means.

Does it mean start the server?
Does it mean start the console?
Does it mean boot the application but doesn't serve any request?

Starting a Rails app has many meaning and it depends in which context you start it. Start to me is mostly boot, but don't serve or allow any console.

**Comment by zzak** [comment]
Ok, that is really good feedback @p8 and @rafaelfranca, I guess "start" can have many meanings and there's a bunch of Rails related processes that can be started [server, console, webpack, sidekiq, etc] -- so I will close this.

Another 💡 is to have a foreman like process manager built into rails that would let you "start" all of these things at once. For me, it is annoying to have to remember to open up another tab and start webpack if I want to run the app in full. 🤔

**Pull request closed without merging**